### PR TITLE
Compatible non SI units

### LINKED ISSUE #3509: Common definition of not really SI-units
Sort of related to unit-checking, but can be added independently.

There are some common units used with the SI-standard, that are not SI-units.
I believe it would be good if we have a common understanding of them, and currently the specification only mentions that base and derived are allowed - but not these ones. In particular it might be good to state that it is only recommended to use them as display units.

First we have the  non SI-units that are acceptable to use with the SI-standard (according to the standard!), and I would in particular want us to consider:
- min (1 minute=60 s)
- h (1 hour=3600 s)
- d (1 day=86400 s)
- l (1 litre=1 dm3)
- eV (1 electronvolt=1.602176634e-19 J)
- (superscript o) spelled "deg" (degree=pi/180 rad)

Additionally the list contains - we might include some or all of them:
- au (astronomical unit=149 597 870 700 m)
- ′ (minute=1/60 deg)
- '' (second = 1/60 minute); in astronomy spelled "as" as in "mas", "μas", "pas"
- ha (hectare = 1e4 m2); I have no idea why it was kept but "are" was dropped.
- t (tonne =1e3 kg)
- Da (dalton = 1.660 539 066 60(50)e-27 kg)
- Np (neper)
- B (bel)
- dB (decibel)

Notes:
- I find the spelling of ' for minute and '' for second as problematic, as it can also mean feet and inches.
- The decibel is problematic as the SI-standard clearly define that dB is for `10*log10(A1/A0)` - whereas MSL has `20*log10(...) ` (We had that discussion at some point).
- I'm not 100% clear on the exact status of degree Celsius (for non-differences), but clearly it is in use in practice.

And then there are the ones that aren't really acceptable according to the standard, but still used and have a clear definition:
- debye (1 qC.m=0.299792458 debye, IIRC)
- English ones like: degree Rankine (=5/9 K) and Fahrenheit (same with offset)
- there are also other ones such as "Ångström" =1e-10 m, but since we want English letters it doesn't fit.

### PR DESCRIPTION
**Pull request by HansOlsson**
According to decisions.
Closes #3509

### CODE REVIEW

File: chapters/unitexpressions.tex [hunk 1]
```diff
@@ -65,10 +65,24 @@ \section{The Syntax of Unit Expressions}\label{the-syntax-of-unit-expressions}
...
+
+The units compatible with the SI system that should be recognized are:
+\begin{itemize}
+\item minute "min" (1 minute=60 s)
+\item hour "h" (1 hour=3600 s)
+\item day "d" (1 day=86400 s)
+\item litre "l" and "L" (1 litre=1 dm3)
+\item electronVolt "eV" (1 electronvolt=1.602176634e-19 J)
+\item degree "deg" (1 degree=pi/180 rad)
+\item debey "debey" (1 debey = 1e-21/299792458 Cm)
```
Comment (maltelenz) [inline chapters/unitexpressions.tex:82]:
```suggestion
\item debye "debye" (1 debye = 1e-21/299792458 Cm)
```

File: chapters/unitexpressions.tex [hunk 2]
```diff
@@ -65,10 +65,24 @@ \section{The Syntax of Unit Expressions}\label{the-syntax-of-unit-expressions}
...
 
+It is recommended that non SI units are only used for the displayUnit attribute -- that ensures that the model is valid even if the unit is not recognized.
+
+The units compatible with the SI system that should be recognized are:
+\begin{itemize}
+\item minute "min" (1 minute=60 s)
+\item hour "h" (1 hour=3600 s)
+\item day "d" (1 day=86400 s)
+\item litre "l" and "L" (1 litre=1 dm3)
+\item electronVolt "eV" (1 electronvolt=1.602176634e-19 J)
```
Comment (maltelenz) [inline chapters/unitexpressions.tex:80]:
```suggestion
\item electron volt "eV" (1 electron volt=1.602176634e-19 J)
```
or `electronvolt`?

File: chapters/unitexpressions.tex [hunk 3]
```diff
@@ -65,10 +65,24 @@ \section{The Syntax of Unit Expressions}\label{the-syntax-of-unit-expressions}
...
 Similarly degree is spelled out as \lstinline!"deg"!, both on its own (for angles) and as part of \lstinline!"degC"!, \lstinline!"degF"! and \lstinline!"degRk"! for temperatures (Celsius, Fahrenheit and Rankine).
 
+It is recommended that non SI units are only used for the displayUnit attribute -- that ensures that the model is valid even if the unit is not recognized.
+
+The units compatible with the SI system that should be recognized are:
+\begin{itemize}
+\item minute "min" (1 minute=60 s)
+\item hour "h" (1 hour=3600 s)
+\item day "d" (1 day=86400 s)
+\item litre "l" and "L" (1 litre=1 dm3)
```
Comment (maltelenz) [inline chapters/unitexpressions.tex:79]:
```suggestion
\item liter "l" and "L" (1 liter=1 dm3)
```
American English, as the style guide suggests

File: chapters/unitexpressions.tex (hunk 2 above)
Comment (HansOlsson) [reply chapters/unitexpressions.tex:80]: The SI Brochure calls it "electronvolt" so I assume that should be used.
```suggestion
\item electronvolt "eV" (1 electronvolt=1.602176634e-19 J)
```

File: chapters/unitexpressions.tex [hunk 4]
```diff
@@ -65,10 +65,24 @@ \section{The Syntax of Unit Expressions}\label{the-syntax-of-unit-expressions}
    NON-DIGIT
 \end{lstlisting}
 
-It is required that basic and derived units of the SI system are recognized, but tools are allowed to additionally support user-defined unit symbols.
+It is required that basic and derived units of the SI system are recognized and in addition some units compatible with the SI system listed below, but tools are allowed to additionally support user-defined unit symbols.
```
Comment (henrikt-ma) [inline chapters/unitexpressions.tex:68]: That was a bit hard to parse.  How about this?
```suggestion
The units required to be recognized are the basic and derived units of the SI system, as well as some units compatible with the SI system listed below, but tools are allowed to additionally support user-defined unit symbols.
```

File: chapters/unitexpressions.tex [hunk 5]
```diff
@@ -65,10 +65,24 @@ \section{The Syntax of Unit Expressions}\label{the-syntax-of-unit-expressions}
...
 The required unit symbols do not make use of Greek letters, but a unit such as $\Omega$ is spelled out as \lstinline!"Ohm"!.
 Similarly degree is spelled out as \lstinline!"deg"!, both on its own (for angles) and as part of \lstinline!"degC"!, \lstinline!"degF"! and \lstinline!"degRk"! for temperatures (Celsius, Fahrenheit and Rankine).
 
+It is recommended that non SI units are only used for the displayUnit attribute -- that ensures that the model is valid even if the unit is not recognized.
```
Comment (henrikt-ma) [inline chapters/unitexpressions.tex:72]: Hyphenate, formatting, and soften _valid_:
```suggestion
It is recommended that non-SI units are only used for the \lstinline!displayUnit! attribute in order to reduce impact of unrecognized unit symbols when using another Modelica tool.
```

File: chapters/unitexpressions.tex [hunk 6]
```diff
@@ -65,10 +65,24 @@ \section{The Syntax of Unit Expressions}\label{the-syntax-of-unit-expressions}
...
+
+The units compatible with the SI system that should be recognized are:
```
Comment (henrikt-ma) [inline chapters/unitexpressions.tex:74]: Avoid weakness of _should_:
```suggestion
The following are the units required to be recognized in addition to the SI system:
```

File: chapters/unitexpressions.tex [hunk 7]
```diff
@@ -65,10 +65,24 @@ \section{The Syntax of Unit Expressions}\label{the-syntax-of-unit-expressions}
...
+\begin{itemize}
+\item minute "min" (1 minute=60 s)
+\item hour "h" (1 hour=3600 s)
+\item day "d" (1 day=86400 s)
+\item liter "l" and "L" (1 liter=1 dm3)
+\item electronvolt "eV" (1 electronvolt=1.602176634e-19 J)
+\item degree "deg" (1 degree=pi/180 rad)
+\item debye "debye" (1 debye = 1e-21/299792458 Cm)
```
Comment (henrikt-ma) [inline chapters/unitexpressions.tex:82]: Formatting:
```suggestion
\item minute \lstinline!"min"! (1 minute = 60 s)
\item hour \lstinline!"h"! (1 hour = 3600 s)
\item day \lstinline!"d"! (1 day = 86400 s)
\item liter \lstinline!"l"! and \lstinline!"L"! (1 liter = 1 dm^{3})
\item electronvolt \lstinline!"eV"! (1 electronvolt = 1.602176634e-19 J)
\item degree \lstinline!"deg"! (1 degree = $\pi/180$ rad)
\item debye \lstinline!"debye"! (1 debye = 1e-21 / 299792458 Cm)
```

File: chapters/unitexpressions.tex [hunk 8]
```diff
@@ -65,10 +65,24 @@ \section{The Syntax of Unit Expressions}\label{the-syntax-of-unit-expressions}
...
+\end{itemize}
+The first 7 are listed as non SI units that are acceptable to use with the SI standard in the SI standard.
```
Comment (henrikt-ma) [inline chapters/unitexpressions.tex:84]: Hyphenate and reorder:
```suggestion
The first 7 are listed in the SI standard as non-SI units that are acceptable to use with the SI system.
```

File: chapters/unitexpressions.tex (hunk 7 above)
Comment (henrikt-ma) [reply chapters/unitexpressions.tex]: I think it would be even better to use Modelica unit strings when defining relation to other units.  For example:
- 1 \lstinline!"min"! = 60 \lstinline!"s"!
- 1 \lstinline!"h"! = 60 \lstinline!"min"!

Maybe one should even express these relations in math mode…

File: chapters/unitexpressions.tex [hunk 9]
```diff
@@ -65,10 +65,24 @@ \section{The Syntax of Unit Expressions}\label{the-syntax-of-unit-expressions}
...
 Similarly degree is spelled out as \lstinline!"deg"!, both on its own (for angles) and as part of \lstinline!"degC"!, \lstinline!"degF"! and \lstinline!"degRk"! for temperatures (Celsius, Fahrenheit and Rankine).
 
+It is recommended that non-SI units are only used for the \lstinline!displayUnit! attribute in order to reduce impact of unrecognized unit symbols when using another Modelica tool.
+
+The following are the units required to be recognized in addition to the SI system:
+\begin{itemize}
+\item minute \lstinline!"min"! (1 minute = 60 s)
+\item hour \lstinline!"h"! (1 hour = 3600 s)
+\item day \lstinline!"d"! (1 day = 86400 s)
+\item liter \lstinline!"l"! and \lstinline!"L"! (1 liter = 1 dm^{3})
```
Comment (HansOlsson) [inline chapters/unitexpressions.tex:79]:
```suggestion
\item liter \lstinline!"l"! and \lstinline!"L"! (1 liter = 1 $\text{dm}^{3}$)
```

Comment (HansOlsson) [reply chapters/unitexpressions.tex:79]: Think this is needed.

**Review by henrikt-ma — APPROVED** [review]
OK, let's save the formatting of the unit relations for a rainy day.

**Comment by HansOlsson** [comment]
> OK, let's save the formatting of the unit relations for a rainy day.

Based on the current situation I would say: _another_ rainy day...
